### PR TITLE
pkg/quick: add Save() function and other enhancements.

### PR DESCRIPTION
**Pull request by balamurugana**
* Add a new function Save() which saves given configuration into given file.
* Simplify Load() function.
* Remove unused CheckVersion().
* CheckData() is a private function now.
* quick_test.go is part of quick package now.
* minio server uses top level quick.Load() and quick.Save() functions.

## How Has This Been Tested?

`make test`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/config-migrate.go [hunk 1]
```diff
@@ -88,33 +88,34 @@ func migrateConfig() error {
...
-
-	}
+	configFile := filepath.Join(getConfigDir(), "fsUsers.json")
 
-	if cv1.Version == "1" {
-		// Purge old fsUsers.json file
-		configFile := filepath.Join(getConfigDir(), "fsUsers.json")
-		removeAll(configFile)
-		console.Println("Removed unsupported config version ‘1’.")
+	cv1 := &configV1{Version: "1"}
```
Comment (vadmeste) [inline cmd/config-migrate.go]: Looks like setting version here is not useful anymore (qc.Load() doesn't check for version header anymore)

**Comment by vadmeste** [comment]
LGTM except one minor comment.

File: cmd/config-migrate.go (hunk 1 above)
Comment (balamurugana) [reply cmd/config-migrate.go]: Yes. Doesn't check and the check didn't work previously in `Load()`.

**Review by donatello — APPROVED** [review]
:+1:

**Pull request merged**